cmd/cdi-cloner/clonertarget.go

```diff
@@ -145,18 +145,19 @@ func (r *prometheusProgressReader) updateProgress() {
 }
 
 // read total file size from reader, and return the value as an int64
-func readTotal(r io.Reader) int64 {
-	totalScanner := bufio.NewScanner(r)
-	if !totalScanner.Scan() {
-		klog.Errorf("Unable to determine length of file")
-		return -1
-	}
-	totalText := totalScanner.Text()
-	total, err := strconv.ParseInt(totalText, 10, 64)
+func readTotal(r io.Reader) (int64, error) {
+	b := make([]byte, 8)
+
+	n, err := r.Read(b)
 	if err != nil {
 		klog.Errorf("%+v", err)
-		return -1
+		return -1, err
+	}
+	if n != len(b) {
+		// Didn't read all 8 bytes..
+		return -1, errors.New("Didn't read all bytes for size header")
 	}
-	klog.V(1).Infoln(fmt.Sprintf("total size: %s", totalText))
-	return total
+	size := int64(binary.LittleEndian.Uint64(b))
```

Comment:
Is 8 bytes enough to hold the maximum size you want to support?  The largest uint64 is 18446744073709551616 which would take 20 ascii bytes to represent.  

Also, You're parsing a 64 bit unsigned integer and converting it to a 64 bit signed integer.  Aren't you losing half of your precision?  Will the size 2^63+1 (9223372036854775809) be handled correctly?

Reply:
Best we add some testing to this code that to ensure that numbers on the periphery work as expected.  ie. 2^64, 2^32+1, 0, -1, etc.

Reply:
int64 by definition is 8 bytes 8x8=64, I am not sending ascii characters, but the actual bytes, so 8 bytes is enough. Yeah there is no reason I can't use uint64 all the way, let me fix that.

Adding tests to catch edge cases.